### PR TITLE
Typescript and Package updates

### PR DESCRIPTION
**Pull request by Apollon77**
* Updates to TS 5.6
* Adjusts all packages ... basic idea is that all is in what reguklar users need in deps, else general stiff is in the main package ... we need to see if that works

### CODE REVIEW

File: packages/matter.js/package.json [hunk 1]
```diff
@@ -53,15 +53,7 @@
         "@matter.js/node": "*"
     },
     "devDependencies": {
-        "@matter.js/tools": "*",
-        "@matter.js/codegen": "*",
```
Comment (lauckhart) [inline packages/matter.js/package.json:57]: This dependency was legit, src/build.config.ts uses `generateSubmoduleProxy` from codegen

File: packages/main/package.json [hunk 2]
```diff
@@ -53,9 +53,7 @@
         "@matter.js/react-native": "*"
     },
     "devDependencies": {
-        "@matter.js/tools": "*",
-        "@matter.js/codegen": "*",
```
Comment (lauckhart) [inline packages/main/package.json:57]: This one is legit

**Review by lauckhart — APPROVED** [review]
Cool thanks.  The two public packages that with dev deps on codegen were legit, it's used in build to generate the "proxies" for submodule re-exports

File: packages/matter.js/package.json (hunk 1 above)
Comment (Apollon77) [reply packages/matter.js/package.json:57]: yes found that out too now ;) re-adding

File: packages/main/package.json (hunk 2 above)
Comment (Apollon77) [reply packages/main/package.json:57]: kk

**Pull request merged**